Applications/Basic/CaptureHDRLoop/CaptureHDRLoop.cpp

```diff
@@ -0,0 +1,100 @@
+        std::vector<Zivid::Settings> settingsVector;
+
+        size_t iris[3] = { 10U, 20U, 30U };
+        int exposureTime[3] = { 10000, 20000, 30000 };
+
+        for(size_t i = 0; i < 3; ++i)
+        {
+            settingsVector.push_back(
+                Zivid::Settings{}
+                    .set(Zivid::Settings::Iris{ iris[i] })
+                    .set(Zivid::Settings::ExposureTime{ std::chrono::microseconds{ exposureTime[i] } })
+                    .set(Zivid::Settings::Brightness{ 1 })
+                    .set(Zivid::Settings::Gain{ 1 })
+                    .set(Zivid::Settings::Bidirectional{ false })
+                    .set(Zivid::Settings::Filters::Contrast::Enabled::yes)
+                    .set(Zivid::Settings::Filters::Contrast::Threshold{ 5 })
+                    .set(Zivid::Settings::Filters::Gaussian::Enabled::yes)
+                    .set(Zivid::Settings::Filters::Gaussian::Sigma{ 1.5 })
+                    .set(Zivid::Settings::Filters::Outlier::Enabled::yes)
+                    .set(Zivid::Settings::Filters::Outlier::Threshold{ 5 })
+                    .set(Zivid::Settings::Filters::Reflection::Enabled::yes)
+                    .set(Zivid::Settings::Filters::Saturated::Enabled::yes)
+                    .set(Zivid::Settings::BlueBalance{ 1.081 })
+                    .set(Zivid::Settings::RedBalance{ 1.709 }));
+            std::cout << "Frame " << i << " " << settingsVector[i] << std::endl;
+        }
+
+        auto hdrFrame = Zivid::HDR::capture(camera, settingsVector);
+
+        std::cout << "Acquiring HDR source images and combining them into HDR frame (slower)" << std::endl;
+        std::vector<Zivid::Frame> frames;
+
+        for(size_t i = 0; i < 3; ++i)
+        {
+            camera << Zivid::Settings::Iris{ iris[i] }
+                   << Zivid::Settings::ExposureTime{ std::chrono::microseconds{ exposureTime[i] } }
+                   << Zivid::Settings::Brightness{ 1 } << Zivid::Settings::Gain{ 1 }
+                   << Zivid::Settings::Bidirectional{ false } << Zivid::Settings::Filters::Contrast::Enabled::yes
+                   << Zivid::Settings::Filters::Contrast::Threshold{ 5 }
+                   << Zivid::Settings::Filters::Gaussian::Enabled::yes
+                   << Zivid::Settings::Filters::Gaussian::Sigma{ 1.5 }
+                   << Zivid::Settings::Filters::Outlier::Enabled::yes
+                   << Zivid::Settings::Filters::Outlier::Threshold{ 5 }
+                   << Zivid::Settings::Filters::Reflection::Enabled::yes
+                   << Zivid::Settings::Filters::Saturated::Enabled::yes << Zivid::Settings::BlueBalance{ 1.081 }
+                   << Zivid::Settings::RedBalance{ 1.709 };
+
+            frames.emplace_back(camera.capture());
+            std::cout << "Frame " << i << " " << camera.settings() << std::endl;
+        }
+
+        std::cout << "Creating the HDR frame" << std::endl;
+        hdrFrame = Zivid::HDR::combineFrames(begin(frames), end(frames));
```

Comment:
I don't like re-using variables. I think it would be nicer if you used two separate variables here, and then both could also be `const`. Makes it easier to reason about the code when you know stuff won't change, you worry less about the scope, side effects etc.

Reply:
Done